src/NeoSharp.BinarySerialization/Cache/BinarySerializerCache.cs

```diff
@@ -43,43 +41,52 @@ internal static BinarySerializerCache InternalRegisterTypes(Type type)
                 if (Cache.TryGetValue(type, out var cache)) return cache;
 
                 var b = new BinarySerializerCache(type);
-                if (b.Count <= 0) return null;
+                if (b.IsEmpty) return null;
 
                 Cache.Add(b.Type, b);
-                return b;
-            }
-        }
 
-        internal static void InternalRegisterTypeConverter(Type type)
-        {
-            lock (TypeConverterCache)
-            {
-                if (TypeConverterCache.ContainsKey(type)) return;
+                if (!b.Type.IsArray)
+                {
+                    // Register array too
 
-                TypeConverterCache.Add(type, (TypeConverter)Activator.CreateInstance(type));
+                    Type array = b.Type.MakeArrayType();
+                    Cache.Add(array, new BinarySerializerCache(array, new BinaryArraySerializer(array, b)));
+                }
+                return b;
             }
         }
 
+        #endregion
+
         /// <summary>
         /// Type
         /// </summary>
         public readonly Type Type;
         /// <summary>
-        /// Count
+        /// IsEmpty
```

Comment:
IsEmpty => _entries.Any() ?

Reply:
_entries are empty too when the cache use directly a `IBinaryCustomSerializer`

https://github.com/CityOfZion/neo-sharp/pull/125/files#diff-1e2f491d84304a3e8dae375555162e48R88